app/models/manageiq/providers/vmware/cloud_manager/vm/operations.rb

```diff
@@ -1,3 +1,20 @@
 module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations
+  extend ActiveSupport::Concern
+
   include_concern 'Power'
+
+  included do
+    supports :terminate do
+      unsupported_reason_add(:terminate, "The VM is powered on") unless current_state == "off"
```

Comment:
@agrare we have modified the code to resemble the behaviour of VCD. Now, it is not possible to delete a VM that is running (powered on). To this end we had to redefine the `supports_terminate?` by looking at the `current_state` of the VM.

Reply:
Ah I like that, we actually have a helper method for this that you can use, https://github.com/ManageIQ/manageiq/blob/master/app/models/vm/operations/power.rb#L22-L24

Reply:
Thanks @agrare. I’ll make another PR to use this.